Stubbing and stuff

needmoney90 opened the pull request:
Made these changes on a branch, I want to merge them into master.

Commit:
Cosmetics and stubbing

Stub out merkato.kill(), stub out comments for commentless functions, add some spacing

Commit:
Add cosmetic changes, stub out merkato kill

Spacing and stubbing for function comments, as well as stubbing a new function merkato.kill()

mark-antony1 reviewed (commented):
overall looks good. finish implementation before pushing

Commit:
Add scaffolding for new order placement scheme

Code is currently broken

Commit:
More fixes and stuff

Change prints to _debugs, make partials fields on the merkato, start styleguide, and add more scaffolding

mark-antony1 commented on merkato/merkato.py:
```diff
@@ -25,23 +25,45 @@ def __init__(self, configuration, coin, base, spread, bid_reserved_balance, ask_
-        # Create ladders from the bid and ask bidget here
+
+        # Exchanges have a maximum number of orders every user can place. Due
+        # to this, every Merkato has a balance of coins that are not currently
```
change balance to reserve

mark-antony1 commented on merkato/merkato.py:
```diff
@@ -25,23 +25,45 @@ def __init__(self, configuration, coin, base, spread, bid_reserved_balance, ask_
+            funds_available = check_reserve_balances(total_pair_balances, allocated_pair_balances, coin_reserve=ask_reserved_balance, base_reserve=bid_reserved_balance)
+
+            if not funds_available:
+                # Don't create the Merkato. Maybe should raise?
```
this is done in check_reserve_balances it will error and kill

needmoney90 replied:
Do we really want to do that though?

mark-antony1 replied:
Yes, I think that is appropriate, at least for now.

mark-antony1 commented on merkato/merkato.py:
```diff
@@ -54,16 +76,31 @@ def __init__(self, configuration, coin, base, spread, bid_reserved_balance, ask_
+        # Cancel all orders
+        self.cancelrange(ONE_SATOSHI, ONE_BITCOIN) # Technically not all, but should be good enough
+
+        # Remove reserve references in DB
```
shouldn't this remove the relevant merkato from the db and then cancel all orders?

needmoney90 replied:
Due to how we're handling partials, we need to sum up all the partial fills of all active orders before cancelling them, and then remove that amount from the partials balance.

mark-antony1 replied:
Why? If all the orders are cancelled it shouldn't matter. No additional trades will be placed or execute and the merkato will be wiped.

needmoney90 replied:
right of course, you're correct

mark-antony1 commented on merkato/merkato.py:
```diff
@@ -75,20 +112,69 @@ def rebalance_orders(self, new_txes):
                 # print('amount', type(tx['amount']), type(tx[PRICE])) # todo use debug
-                
+
+                if partial_fill:
```
extract everything under partial_fill to a seperate merkato class function

mark-antony1 commented on merkato/merkato.py:
```diff
@@ -75,20 +112,69 @@ def rebalance_orders(self, new_txes):
+                    # 3. Skip everything else
+                    continue
+
+                if tx['id'] in filled_orders:
```
i would extract this to a variable called matching_order_exists or something

mark-antony1 commented on merkato/merkato.py:
```diff
@@ -75,20 +112,69 @@ def rebalance_orders(self, new_txes):
+                # We need to place a matching order
+                # We want to get the total amount of that order
+
+                total_amount = get_total_amount(tx['id']) # todo unimplemented
```
this needs to be implemented

mark-antony1 commented on merkato/merkato.py:
```diff
@@ -115,11 +201,42 @@ def rebalance_orders(self, new_txes):
+                # We need to place a matching order
+                # We want to get the total amount of that order
+
+                total_amount = get_total_amount(tx['id']) # todo unimplemented
```
implement

mark-antony1 commented on merkato/merkato.py:
```diff
@@ -211,6 +334,7 @@ def distribute_bids(self, price, total_to_distribute, step=1.02):
 
         # 4. Store the remainder of total_to_distribute, as well as the final
         #    order placed in decaying_bid_ladder
+        # TODO
```
does this need to be done for this feature?

needmoney90 replied:
No, it's being handled by the decaying ladder itself.

mark-antony1 commented on merkato/merkato.py:
```diff
@@ -397,44 +524,63 @@ def modify_settings(self, settings):
 
+
     def add_reserve(self):
+        ''' TODO: Function comment
```
does this need to be implemented for this feature?

needmoney90 replied:
How do we initially set reserve again?

mark-antony1 replied:
in the merkato init, reserve is inputted by the user and then used to distribute ladders

needmoney90 replied:
This isn't necessary (yet).

mark-antony1 commented on merkato/utils/database_utils.py:
```diff
@@ -59,25 +74,41 @@ def update_merkato(exchange_pair, key, value):
         conn.close()
 
 
+def kill_merkato(exchange_pair):
```
implement

mark-antony1 commented on merkato/utils/database_utils.py:
```diff
@@ -129,98 +172,151 @@ def get_all_exchanges():
 
 
+def remove_base_partial(UUID, amount)
+pass
```
implement

mark-antony1 commented on merkato/utils/database_utils.py:
```diff
@@ -129,98 +172,151 @@ def get_all_exchanges():
+
+
+def remove_quote_partial(UUID, amount)
+pass
```
implement

mark-antony1 commented on merkato/utils/database_utils.py:
```diff
@@ -129,98 +172,151 @@ def get_all_exchanges():
+# I realize this could be done with negative values. For readability's sake,
+# i'm implementing it so you never need to pass in a negative number.
+def add_base_partial(UUID, amount)
+pass
```
implement

mark-antony1 commented on merkato/utils/database_utils.py:
```diff
@@ -129,98 +172,151 @@ def get_all_exchanges():
+
+
+def add_quote_partial(UUID, amount)
+pass
```
implement

Commit:
some fixes

Commit:
More changes

Commit:
More fixes

Commit:
remove one call of our ladder function from distributes

Commit:
add partials to insert

Commit:
Merge branch 'master' into bunching-test

Commit:
Add db calls

Commit:
add merkato killing

Commit:
stubbing

Commit:
more fixes

Commit:
Merge branch 'master' into bunching-test
